lib/gpu/lib/src/context.dart

```diff
@@ -46,6 +46,18 @@ base class GpuContext extends NativeFieldWrapperClass1 {
     return _getMinimumUniformByteAlignment();
   }
 
+  /// Whether the backend supports multisample anti-aliasing for offscreen
+  /// color and stencil attachments. A subset of OpenGLES-only devices do not
+  /// support this functionality.
+  ///
+  /// Any texture created via [createTexture] is an offscreen texture.
+  /// There is currently no way to render directly against the "onscreen"
+  /// texture that the framework renders to, so all Flutter GPU textures are
+  /// "offscreen".
+  bool get doesSupportOffscreenMSAA {
```

Comment:
Just a thought. Maybe this doesn't actually need to be exposed at all.Just let folks construct the same MSAA or non-MSAA topolgy, and if its not supported then ignore the transient attachments and use the resolve attachment.

Practically speaking, the number of devices flutter targets that does not support MSAA is sooo small. I don't know if its worth carrying around in our API.

Additionaly, I don't know if this handles it but we need to do MSAA a bit differently for GLES. We don't ever actually allocate the transient attachment.

Reply:
Oooh, I think for the common case of `kMultisampleResolve`, coercing to `kStore` and ignoring the multisample texture just about works. But unfortunately, things get funky with `kStoreAndMultisampleResolve` and I don't think we can reasonably assume the user's intent.

> Additionaly, I don't know if this handles it but we need to do MSAA a bit differently for GLES. We don't ever actually allocate the transient attachment.

IIRC we did this in a way where it automagically does the right thing when you try to allocate a transient multisample texture on GLES (unless I'm thinking of the wrong thing). :)

If there's special HAL usage surrounding this, could you link to where we're doing it in the 2D renderer to help jog my memory?

Reply:
I'm not sure if it would work automatically or not, see https://github.com/flutter/engine/blob/2d0eac0ea9389bdaff89d230af64536d9ad32fd8/impeller/renderer/render_target.cc#L342

Reply:
I think `SupportsImplicitResolvingMSAA` is just for MSAA against the onscreen texture

Reply:
No, this is used for all offscreen MSAA on the GLES backend.

Reply:
Oh I'm remembering this now. I don't know if I can easily hide this complexity under the API. With the way the HAL currently works, I would have to surface and document `SupportsImplicitResolvingMSAA`. The user is in an alternate reality where they have to creates, attach, and bind fake multisample textures.

Reply:
I'm not sure the right way to fix this in the API, but maybe consider only supporting MSAA on metal/vulkan for now and coming back to GLES?

Reply:
Yeah. I changed this to add a graceful fallback to not use MSAA when "normal" MSAA support is not possible. I kept the check as a renderer could fallback to other AA techniques (such as FXAA) when MSAA is unavailable.